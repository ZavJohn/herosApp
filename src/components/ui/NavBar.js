import React from 'react'
import { Link, NavLink, useNavigate } from 'react-router-dom';

export const Navbar = () => {

    const navigate = useNavigate();
    const handleLogout = () => {
        navigate('/login', {
            replace: true
          });
    };

    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="/"
            >
                Asociaciones
            </Link>

            <div className="collapse navbar-collapse" id="navbarNav">
                <div className="navbar-nav container-fluid">

                    <NavLink 
                        className={ ( {isActive} ) => 'nav-item nav-link btn' +  (isActive ? 'active' : '') }
                        to="/marvel"
                        aria-current="page"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        className={ ( {isActive} ) => 'nav-item nav-link btn' +  (isActive ? 'active' : '') }
                        to="/dc"
                        aria-current="page"
                    >
                        DC
                    </NavLink>
                    <NavLink 
                        className={ ( {isActive} ) => 'nav-item nav-link btn' +  (isActive ? 'active' : '') }
                        to="/search"
                        aria-current="page"
                    >
                        Search
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                <ul className="navbar-nav ml-auto">
                    <span
                        className="nav-item nav-link text-info"
                    > 
                        Jonathan Zavala
                    </span>
                    <button 
                        className="nav-item nav-link btn" 
                        onClick={ handleLogout }
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}