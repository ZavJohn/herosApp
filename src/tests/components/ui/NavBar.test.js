
import { mount } from 'enzyme';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { AuthContext } from '../../../auth/authContext';
import { Navbar } from '../../../components/ui/NavBar';
import { types } from '../../../types/types';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}));

describe('Pruebas en el <NavBar />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Peter',
            logged: true
        }
    };
    
    const wrapper = mount(
        <AuthContext.Provider value= { contextValue }>
            <MemoryRouter initialEntries={[ '/' ]}>
                <Routes>
                    <Route path="/" element={ <Navbar /> } />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    );
    
    test('Debe de mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim() ).toBe('Peter');

    });

    test('AcciÃ³n al Logout y dispatch con los argumentos', () => {

        wrapper.find('button').prop('onClick')();

        expect( contextValue.dispatch ).toHaveBeenCalledWith({ 'type': types.logout })
        expect( mockNavigate ).toHaveBeenCalledWith('/login', { replace: true });

    });
})