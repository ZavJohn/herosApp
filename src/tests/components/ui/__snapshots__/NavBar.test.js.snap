// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <NavBar /> Debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <div
      className="navbar-nav container-fluid"
    >
      <a
        className="nav-item nav-link btn"
        href="/marvel"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        className="nav-item nav-link btn"
        href="/dc"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        className="nav-item nav-link btn"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Peter
      </span>
      <button
        className="nav-item nav-link btn"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
